autogen/agentchat/contrib/capabilities/context_handling.py

```diff
@@ -0,0 +1,46 @@
+from typing import Dict, Optional, Union, List, Tuple, Any
+from autogen.agentchat.contrib.capabilities.agent_capability import AgentCapability
+from autogen import ConversableAgent
+
+
+class LongContextCapability:
```

Comment:
Could be called "ShortTermMemoryCapability" as is more sensational :D 

Another capability can be called "LongTermMemoryCapability" which selects "pinned" messages and a set of RAG-selected messages.

Reply:
Long-context handling is certainly related to memory, in the sense that context is a form of memory broadly speaking. But if we use the term memory in too many different settings, it will become less useful to us as a technical term. I suggest reserving _memory_ for information stored using vector databases, such as the memory used for teachable agents and the learning agents we are working on. I wouldn't use the term _RAG_ at all in these settings because that term implies too much about where the data comes from, like a set of files or the web.

In summary, I think _LongContext_ and _LongContextHandling_ are ideal descriptors of what this new capability is about.

Reply:
Or we just call it as it is: SlidingContextWindow ...

Reply:
This initial implementation is limited to a sliding context window, but I think @gagb plans to quickly add options for more sophisticated context handling.

Reply:
I agree the term "memory" should be reserved though "context" is somewhat ambiguous. Perhaps `LongChatHistoryHandling` is more accurate. If we have multiple capability classes for different long context handling options, then `LongChatHistoryHandling` by itself is not accurate enough. We could use `SlidingWindowChatHistoryHandling` for example.

Reply:
I think context almost always refers to the transformer's context window when discussing LLMs. 

As @gagb said at the top, he will extend this capability with configurable support for other ways of handling long context, instead of implementing each as a different capability.

Reply:
Okay I am sold long context is fine.

Reply:
I found an even better name that I had been thinking for a while -- Transform chat history. We should think about many of these operations in terms of transformations. It's a commonly used term in other libraries (eg scikit) and it actually captures the operations. Verbs like handling are more vague.

```python
assistant = autogen.AssistantAgent("assistant")
transform_chat_history = TransformChatHistory(max_tokens_per_message=50, max_messages=10, max_tokens=1000)
transform_chat_history.add_to_agent(assistant)
```

Reply:
Please resolve this thread if you are happy with the name.